sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -42,13 +44,16 @@ enum class PollSocketDirection
  * @param socketFileDescriptor socket descriptor.
  * @param direction poll events for read or write socket.
  * @param timeout  return if polling for more than \p timeout
+ * @param context The context while polling that can be use to cancell waiting for socket.
+ *
  * @return int with negative 1 upon any error, 0 on timeout or greater than zero if events were
  * detected (socket ready to be written/read)
  */
 int pollSocketUntilEventOrTimeout(
     curl_socket_t socketFileDescriptor,
     PollSocketDirection direction,
-    long timeout)
+    long timeout,
```

Comment:
Do we have to use long and have the ms precision, or can this be in seconds?

Reply:
we need ms,

Reply:
Can you clarify why? What's requiring ms?

Reply:
sure. For Linux, we use `poll()` https://man7.org/linux/man-pages/man2/poll.2.html
and for Windows, we use `WSAPoll()` https://docs.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-wsapoll
for waiting for a socket to be ready..

They take an int parameter for the timeout which defines miliseconds.

Think I just need to update from long to int..  It's not too bad right now becasue the timeout is hardcoded by us to not override the int with the long

Reply:
Internally, as an implementation detail of this method we could always take the time in seconds (as int), mutiply it by 1000 and pass it to poll/WSAPoll as a long (in ms). 

The question is, will the caller of this API ever want to control the timeout in ms precision? It doesn't look like we need that. We seem to be passing 60s, (`60000L`), so why not just accept seconds?